### PR TITLE
add integration tests for State transitions for a failure in backup restore

### PR DESCRIPTION
**Pull request by meldafrawi**
https://github.com/longhorn/longhorn/issues/1260

### CODE REVIEW

**Comment by yasker** [comment]
LGTM.

File: manager/integration/tests/common.py [hunk 1]
```diff
@@ -3343,3 +3343,52 @@ def settings_reset():
...
+            backup_target_setting = client.update(backup_target_setting,
+                                                  value=backupsettings[0])
+            assert backup_target_setting.value == backupsettings[0]
+
+            backup_target_credential_setting = client.by_id_setting(
+                SETTING_BACKUP_TARGET_CREDENTIAL_SECRET)
+            backup_target_credential_setting = \
+                client.update(backup_target_credential_setting,
+                              value=backupsettings[1])
+            assert backup_target_credential_setting.value == backupsettings[1]
```
Comment (yasker) [inline manager/integration/tests/common.py:3388]: We should `break` after found the s3 backup target. Same below.

**Comment by meldafrawi** [comment]
Yes, `add wait replica fail for crash_replica_processes` commit should be the second to last.

File: manager/integration/tests/common.py (hunk 1 above)
Comment (meldafrawi) [reply manager/integration/tests/common.py:3388]: fixed

**Pull request merged**